cupy/cuda/memory.pyx

```diff
@@ -799,6 +937,29 @@ cdef class PooledMemory(BaseMemory):
         self.device_id = chunk.mem.device_id
         self.pool = pool
 
+        # we need a way to know what the underlying memory is
+        # TODO(leofang): would it be better to do this in MemoryPointer?
+        if isinstance(chunk.mem, Memory):
+            self.identity = "Memory"
+        elif isinstance(chunk.mem, MemoryAsync):
+            self.identity = "MemoryAsync"
+        elif isinstance(chunk.mem, UnownedMemory):
+            self.identity = "UnownedMemory"
+        elif isinstance(chunk.mem, SystemMemory):
+            self.identity = "SystemMemory"
+            self.prefetch = <SystemMemory>(chunk.mem).prefetch
+            self.advise = <SystemMemory>(chunk.mem).advise
```

Comment:
Q: Who calls these functions (`prefetch` and `advise`)?

Reply:
xref: https://github.com/cupy/cupy/pull/8442#discussion_r1774360929

I had `prefetch`/`advise` added when experimenting the support but I commented them out later due to performance degradation (it adds too much overhead to small arrays). The comment was removed in commit f618bab75.

Reply:
Let me know if you prefer to not expose this for now.

Reply:
Thanks, got it. Let's keep it as is.